src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -1757,20 +1748,30 @@ public static Design routeDesignWithUserDefinedArguments(Design design, String[]
 	 * @param config A {@link RWRouteConfig} instance consisting of customizable parameters to use.
 	 */
 	private static Design routeDesign(Design design, RWRouteConfig config) {
+		return routeDesign(design, config, () -> {
+			if(config.isPartialRouting()) {
+				return config.isResolveConflictNets()? new RapidStreamRoute(design, config) : new PartialRouter(design, config);
```

Comment:
This is still wrong, should be: https://github.com/eddieh-xlnx/RapidWright/blob/716e796bb799d6dd3f327e3f382b0e2a3c6e50f3/src/com/xilinx/rapidwright/rwroute/RWRoute.java#L1753

As a suggestion for the future collaboration, please cherry-pick contributions from others as that can help prevent mistakes and preserves attribution.

Reply:
I made that change on purpose, as the suggested seems to have a problem in instantiating the correct router object for the specific use case.

Reply:
Thanks for the further comment making it clear. This is updated in the new commit.